pkg/cloud/cloud_io.go

```diff
@@ -147,6 +154,34 @@ func IsResumableHTTPError(err error) bool {
 		sysutil.IsErrConnectionRefused(err)
 }
 
+// ResumingReaderRetryOnErrFnForSettings returns a function that can
+// be passed as a RetryOnErrFn to NewResumingReader.
+func ResumingReaderRetryOnErrFnForSettings(
+	ctx context.Context, st *cluster.Settings,
+) func(error) bool {
+	return func(err error) bool {
+		if IsResumableHTTPError(err) {
+			return true
+		}
+
+		retryTimeouts := retryConnectionTimedOut.Get(&st.SV)
+		// We could test this error specifically for ETIMEOUT
+		// by grabbing the Errno:
+		//
+		//    err.(*net.OpError).Err.(*os.SyscallError).Err.(syscall.Errno)
+		//
+		// But that assumes that those error types surive all
+		// the way here along all call paths.
+		//
+		// For now, we test for the string instead.
+		if retryTimeouts && strings.Contains(err.Error(), "connection timed out") {
```

Comment:
I'm trying to write a good test to confirm if we can just do `errors.Is(err, syscall.ETIMEDOUT)`